core/src/main/java/io/kestra/core/schedulers/AbstractScheduler.java

```diff
@@ -147,33 +149,57 @@ public void run() {
         );
         thread.start();
 
-        // remove trigger on flow update & update local triggers store
+        // remove trigger on flow update, update local triggers store, and stop the trigger on the worker
         this.flowListeners.listen((flow, previous) -> {
             if (flow.isDeleted() || previous != null) {
-                List<AbstractTrigger> triggersDeleted = flow.isDeleted() ? ListUtils.emptyOnNull(flow.getTriggers()) : FlowService
-                    .findRemovedTrigger(flow, previous);
+                List<AbstractTrigger> triggersDeleted = flow.isDeleted() ?
+                    ListUtils.emptyOnNull(flow.getTriggers()) :
+                    FlowService.findRemovedTrigger(flow, previous);
+
                 triggersDeleted.forEach(abstractTrigger -> {
                     Trigger trigger = Trigger.of(flow, abstractTrigger);
                     this.triggerQueue.delete(trigger);
+
+                    this.executionKilledQueue.emit(ExecutionKilledTrigger
+                        .builder()
+                        .tenantId(trigger.getTenantId())
+                        .namespace(trigger.getNamespace())
+                        .flowId(trigger.getFlowId())
+                        .triggerId(trigger.getTriggerId())
+                        .build()
+                    );
```

Comment:
You also kill/remove polling triggers that are currently evaluated?

I wonder if this behavior will be visible to the user like failed execution created by the Worker or late/missed evaluation.

Reply:
Yes it will be included, and it will send an error log for InterruptedException. I've decided to let this visible for now. 
In a futur, maybe we can capture and hide the log, but I really think for now, we need to keep the log to help people know that something was stopped (maybe in the middle of something)